qiime2/sdk/tests/test_usage.py

```diff
@@ -380,9 +390,28 @@ def test_init_data(self):
         with self.assertRaisesRegex(TypeError, 'expected Metadata'):
             use.init_metadata('name', lambda: object)
 
+        with self.assertRaisesRegex(ValueError, 'expected a ScopeRecord.'):
+            use.init_data_collection('', list, object)
+
+        with self.assertRaisesRegex(ValueError, 'expected a ScopeRecord.'):
+            use.init_data_collection('', list,
+                                     usage.ScopeRecord('', object, ''), object)
+
     def test_merge_metadata(self):
         use = usage.ExecutionUsage()
         md1 = use.init_metadata('md1', examples.md1_factory)
         md2 = use.init_metadata('md2', examples.md2_factory)
         merged = use.merge_metadata('md3', md1, md2)
         self.assertIsInstance(merged.result, Metadata)
+
+    def test_variadic_input_simple(self):
+        use = usage.ExecutionUsage()
+        action = self.plugin.actions['variadic_input_method']
+        action.examples['variadic_input_simple'](use)
+        for record in use._get_records().values():
```

Comment:
Rather than iterating through, can you specifically look up the inputs and the outputs?

```python
int_list_record = use.get_result('int')
int_set_record = use.get_result('int_set')
out_record = use.get_result('out')

...
```

My concern here is that the loop doesn't explicitly check for our _known_ records. This is a testing pitfall that I have gotten sucked into before...

Reply:
Not sure why, but `Usage.get_result` only works if `record.source == 'action'`.  How does 23ef3a5 this look?

Reply:
That's right, I forgot that `get_result` is "protected".

The tests in 23ef3a5 look good, but you're still effectively looping over the scope records, rather than explicitly evaluating them one by one.

```python
int_list_record = use._get_record('int')
...
```

Make sense?

Reply:
Alternatively, you could stick with the unpacking, and add in additional assertions about the scoperecord's refs:

```python
self.assertEqual(ints_a.ref, 'ints_a')
self.assertIsInstance(ints_a.result, Artifact)
...
```